src/qbe_ir_generator.cpp

```diff
@@ -151,6 +151,35 @@ void QbeIrGenerator::Visit(const IfStmtNode& if_stmt) {
   output << end_label << std::endl;
 }
 
+void QbeIrGenerator::Visit(const WhileStmtNode& while_stmt) {
+  int label_num = NextLabelNum();
+  std::string body_label = PrefixLabel("loop_body", label_num);
+  std::string pred_label = PrefixLabel("pred", label_num);
+  std::string end_label = PrefixLabel("end", label_num);
+  int predicate_num = 0;
```

Comment:
```suggestion
```
The scope of this variable is larger than necessary. It is utilized within two distinct conditional blocks, each with its own lifetime. Therefore, it is considered good practice to declare the variable separately within each of the two conditional blocks.